addons/im_livechat/tests/test_get_mail_channel.py

```diff
@@ -138,3 +141,24 @@ def test_operator_livechat_username(self):
         self.assertEqual(message_formats[0]['author_id'][1], operator.livechat_username)
         self.assertEqual(message_formats[0]['author_id'][2], operator.livechat_username)
         self.assertFalse(message_formats[0].get('email_from'), "should not send email_from to livechat user")
+
+    def test_read_channel_unpined_for_operator_after_one_day(self):
+        public_user = self.env.ref('base.public_user')
+        channel_info = self.livechat_channel.with_user(public_user)._open_livechat_mail_channel(anonymous_name='visitor')
+        operator_channel_partner = self.env['mail.channel.partner'].search([('channel_id', '=', channel_info['id']), ('partner_id', 'in', self.operators.partner_id.ids)])
+        message = self.env['mail.channel'].browse(channel_info['id']).message_post(body='cc')
+        operator_channel_partner.channel_id.with_user(self.operators.filtered(
+            lambda operator: operator.partner_id == operator_channel_partner.partner_id
+        ))._channel_seen(message.id)
```

Comment:
could use `operator_channel_partner.channel_id` instead of filter I think

Reply:
Here we will need to use `with_user`, because of the same different user issue.